controlplane/kubeadm/internal/filters.go

```diff
@@ -309,6 +309,9 @@ func matchInitOrJoinConfiguration(machineConfig *bootstrapv1.KubeadmConfig, kcp
 
 	// takes the KubeadmConfigSpec from KCP and applies the transformations required
 	// to allow a comparison with the KubeadmConfig referenced from the machine.
+	if kcp.Spec.KubeadmConfigSpec.InitConfiguration == nil {
```

Comment:
Given we do this before deepcopy this will propagate back up and end up in defer patch, right? (not sure if that leads to an actual patch).

We can move this to after the deepcopy in getAdjustedKcpConfig, but I wonder if instead there is a way to cleanup the code in cleanupConfigFields (l.394-398)

Reply:
fixed, TPAL